include/bbp/sonata/report_reader.h

```diff
@@ -166,17 +166,25 @@ class SONATA_API ReportReader
         Population(const H5::File& file, const std::string& populationName);
         std::pair<size_t, size_t> getIndex(const nonstd::optional<double>& tstart,
                                            const nonstd::optional<double>& tstop) const;
+        /**
+         * Return the ElementIds for the given selection, alongside the filtered NodePointers
+         * and the range of positions where they fit in the file. This latter two are necessary
+         * for performance to understand how and where to retrieve the data from storage.
+         *
+         * \param node_ids limit the report to the given selection. If nullptr, all nodes in the
+         * report are used
+         */
+        std::tuple<NodePointers, Range, typename DataFrame<KeyType>::DataType> getElementIds(
+            const nonstd::optional<Selection>& node_ids = nonstd::nullopt) const;
 
-        std::map<NodeID, Range> nodes_pointers_;
+        NodePointers nodes_pointers_;
```

Comment:
As big as the number of GIDs times the ranges in the file (i.e., 3 x `uint64_t` per GID). This data structure is required and has not changed compared to the original implementation.

Keep in mind that, even when using a very big report like the reference 60TB, it would be around 100MB of memory in total (4234929 x 3 x 8 = 96.9MB). I'll reply to your other comment below, but the idea of this PR is to re-use the same data structure instead of splitting the `NodeID`s in a vector and then the positions in another vector, like in the previous implementation.